plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1499,7 +1499,10 @@ public void destroyPatchVbd(final Connection conn, final String vmName) throws X
                 final Set<VBD> vbds = vm.getVBDs(conn);
                 for (final VBD vbd : vbds) {
                     if (vbd.getType(conn) == Types.VbdType.CD) {
-                        vbd.eject(conn);
+                        if (!vbd.getEmpty(conn)) {
```

Comment:
@rhtyd Makes sense to eject only when ISO is inserted. Verified that this is being handled in all cases before ejecting vdb, except here.

Reply:
agree that it makes sense, but I don't see why this yields the bug @vladimirpetrov saw. However if this fixes it for him, i'm all :+1: